### PR TITLE
Issue forms

### PR DESCRIPTION
**Pull request by sagikazarmark**
This PR demonstrates a couple other options for issues provided by GitHub:

An alternative to the markdown issue templates is using issue forms. It's a nice way to make bug reports more sophisticated (eg. make sure to include a version field).

This PR converts the current markdown templates to issue forms. Preview:

- https://github.com/sagikazarmark/kit/blob/issue-forms/.github/ISSUE_TEMPLATE/bug_report.yaml
- https://github.com/sagikazarmark/kit/blob/issue-forms/.github/ISSUE_TEMPLATE/feature_request.yaml

It also adds a couple links to the issue opening page offering additional help: https://github.com/sagikazarmark/kit/blob/issue-forms/.github/ISSUE_TEMPLATE/config.yml

TBH, issue forms is not necessarily better for the _current templates_, because they are quite simple.

A couple examples from other projects:
- [Dex bug report](https://github.com/dexidp/dex/blob/master/.github/ISSUE_TEMPLATE/bug_report.yaml)
- [Dex feature request](https://github.com/dexidp/dex/blob/master/.github/ISSUE_TEMPLATE/feature_request.yaml)
- [Dex new issue](https://github.com/dexidp/dex/issues/new/choose)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: 🐛 Bug report
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:1]:
```suggestion
name: Bug report
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+description: Report a bug to help us improve
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:2]:
```suggestion
description: Report a bug
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+blank_issues_enabled: false
+contact_links:
+  - name: ❓ Ask a question
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:3]:
```suggestion
  - name: Ask a question
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    url: https://github.com/go-kit/kit/discussions/new?category=q-a
+    about: Ask and discuss questions with other community members
+
+  - name: 🌐 Website
+    url: https://gokit.io/
+    about: Check the website for details
+
+  - name: 📓 Reference
+    url: https://pkg.go.dev/mod/github.com/go-kit/kit
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:12]:
```suggestion
    url: https://pkg.go.dev/github.com/go-kit/kit
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+blank_issues_enabled: false
+contact_links:
+  - name: ❓ Ask a question
+    url: https://github.com/go-kit/kit/discussions/new?category=q-a
+    about: Ask and discuss questions with other community members
+
+  - name: 🌐 Website
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:7]:
```suggestion
  - name: Website
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    url: https://gokit.io/
+    about: Check the website for details
+
+  - name: 📓 Reference
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:11]:
```suggestion
  - name: Reference
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    url: https://pkg.go.dev/mod/github.com/go-kit/kit
+    about: Check the Go code reference
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:13]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+  - name: 💬 Slack channel
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:15]:
```suggestion
  - name: Slack channel
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    url: https://gophers.slack.com/messages/go-kit
+    about: Please ask and answer questions here
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:17]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+blank_issues_enabled: false
+contact_links:
+  - name: ❓ Ask a question
+    url: https://github.com/go-kit/kit/discussions/new?category=q-a
+    about: Ask and discuss questions with other community members
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:5]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+
+  - name: 🌐 Website
+    url: https://gokit.io/
+    about: Check the website for details
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/config.yml:9]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+name: 🚀 Feature request
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:1]:
```suggestion
name: Feature request
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+description: Suggest an idea for this project
```
Comment (peterbourgon) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:2]:
```suggestion
description: Suggest new functionality or an enhancement
```

**Review by peterbourgon — APPROVED** [review]
Oh yeah, didn't know about this. Nice!

**Comment by sagikazarmark** [comment]
@peterbourgon apparently contact links don't work without an about text. :\ What would you like me to do?

**Comment by peterbourgon** [comment]
Dumb. Just put them back I guess.

**Pull request merged**